projectmanagement/src/main/java/life/qbic/projectmanagement/ProjectIntent.java

```diff
@@ -0,0 +1,14 @@
+package life.qbic.projectmanagement;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * A project intent contains information on the project that is related to the intent of the
+ * project.
+ */
+public record ProjectIntent(ProjectTitle projectTitle) {
```

Comment:
Afaik, records introduce public constructors, which I would like to avoid to create expressive object APIs with named static factory methods.

Reply:
I think that `records` here are perfect for implementing value objects.
https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/Record.html
Records have components (in this case a ProjectTitle). As every component needs to be provided for a record to be complete (or in this case a project intent). You can, however, write custom constructors or provide static factory methods. For value objects, we need to provide all components. Every internal value of the value object must be verified in any case, for this the compact constructors of records are perfect IMHO.

Reply:
We further benefit from the guaranteed correct implementation of `equals and hashCode` as well as the `toString` and immutability of java records

Reply:
@KochTobi ok, convinced. But let's keep an eye on the object APIs